### PR TITLE
Add metrics crud json fields

### PR DESCRIPTION
**Pull request by jessie-ross**
**This PR adds JSON fields to Metric Create/Update.**

This is kind of an intermediate PR before the completion in #329.

Part of a series on #13 and #14.

## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)

### CODE REVIEW

**Comment by jessie-ross** [comment]
(Not sure why but I have had to rerun this more than 5 times on CircleCI to get it working, usually timing out on completely unrelated code.)

File: pages/metrics/index.tsx [hunk 1]
```diff
@@ -114,7 +115,8 @@ const MetricsIndexPage = () => {
             {(formikProps) => (
               <form onSubmit={formikProps.handleSubmit}>
                 <DialogContent>
-                  <MetricFormFields formikProps={formikProps as FormikProps<{ metric: unknown }>} />
+                  {/* This unknown case is temporary to help with the split */}
```
Comment (jessie-ross) [inline pages/metrics/index.tsx:118]: The splitting of this PR I mean, resolved in #329

**Review by aaronyan — APPROVED** [review]
LGTM! Thanks for adding this in time for us to help experimenters add metrics (when/if they request them!)

**Comment by aaronyan** [comment]
> (Not sure why but I have had to rerun this more than 5 times on CircleCI to get it working, usually timing out on completely unrelated code.)

Yeah, that's frustrating. I've had some similar experiences with the CircleCI tests, and maybe worth looking into. Creating an issue for now! https://github.com/Automattic/abacus/issues/331

**Pull request merged**